### PR TITLE
chore: add support for multi platform docker images

### PR DESCRIPTION
**Pull request by rskrishnar**
## Description of the change

Builds multi-platform docker images for rudder-transformer (also for the user transformer).

Support has been added for both CodeBuild, and GH actions build configurations for backward compatibility. We'll decommission the CodeBuild pipelines once these changes are released into production.

Additionally, the images are also pushed into `rudderstack` organization along with `rudderlabs` in the DockerHub. Once the necessary changes are done to the public documentation and all the production deployments are migrated, we'll stop pushing images to the old repositories.

Moreover, only a single set of DockerHub credentials are used everywhere now.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

N/A

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [x]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: buildspec.saas.yml [hunk 1]
```diff
@@ -26,11 +26,17 @@ phases:
...
     commands:
-      - docker push rudderlabs/rudder-transformer:$VERSION
-      - docker push rudderlabs/rudder-transformer:latest
+      - VERSION="$(date '+%d%m%Y.%H%M%S')"
+
+      # creates a docker builder instance to build multi-platform docker images
+      - docker buildx create --use
+
+      # builds and pushes multi-platform docker images to docker registry
+      - docker buildx build --build-arg version=${VERSION} --platform=linux/amd64,linux/arm64 -t rudderlabs/rudder-transformer:latest -t rudderlabs/rudderstack-transformer:$VERSION -f Dockerfile --push .
```
[the remark this answers is not in the document]
Comment (sanpj2292) [reply buildspec.saas.yml:37]: I just see two tags being built for same image.
Sorry if I am mis-reading something here but can you point me to the location where you think the image building is happening twice ?

[the remark this answers is not in the document]
Comment (saikumarrs) [reply buildspec.saas.yml:37]: Ln 25 and Ln 37.

[the remark this answers is not in the document]
Comment (rskrishnar) [reply buildspec.saas.yml:37]: when we build images using buildx, the built images won't show up in `docker images`. we can only push them to docker registry

if we want the images to show up, we need to run a separate command and this command doesn't work everywhere.

**Review by saikumarrs — DISMISSED** [review]

**Comment by utsabc** [comment]
@rskrishnar can you share some context around this, I am not well versed in multi platform docker images

**Review by sanpj2292 — APPROVED** [review]

**Comment by saikumarrs** [comment]
Closing this PR as it is not a priority now.

**Review by utsabc — APPROVED** [review]

**Comment by saikumarrs** [comment]
I have resurrected this PR by extending the support for all the build configurations.

**Comment by koladilip** [comment]
@saikumarrs right we are using both flows AWS codebuild and github?

**Comment by saikumarrs** [comment]
> @saikumarrs right we are using both flows AWS codebuild and github?

Yeah, only for now. After the next release, AWS CodeBuild pipelines will be deleted.

**Comment by koladilip** [comment]
@saikumarrs Post-build, please verify that those images are working as expected then we can merge the PR

**Review by koladilip — APPROVED** [review]

File: .github/workflows/master-bulids.yml [hunk 2]
```diff
@@ -4,9 +4,64 @@ on:
     branches:
       - 'master'
 env:
-  REPO_NAME: rudderlabs/develop-rudder-transformer
+  REPO_NAME_OLD: rudderlabs/develop-rudder-transformer
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply .github/workflows/master-bulids.yml:7]: Yes, temporary.

**Comment by saikumarrs** [comment]
> @saikumarrs Post-build, please verify that those images are working as expected then we can merge the PR

The latest images have been tested and everything looks good.
image

File: buildspec.dev.yml [hunk 3]
```diff
@@ -4,6 +4,13 @@ phases:
   install:
     runtime-versions:
       nodejs: 14
+    commands:
+      - 'echo "${DOCKERHUB_PASSWORD}" | docker login --username rudderlabs --password-stdin'
+      - wget https://github.com/docker/buildx/releases/download/v0.10.2/buildx-v0.10.2.linux-amd64
+      - mkdir -p ~/.docker/cli-plugins
+      - mv buildx-v0.10.2.linux-amd64 ~/.docker/cli-plugins/docker-buildx
+      - chmod a+rx ~/.docker/cli-plugins/docker-buildx
+      - docker run --privileged --rm tonistiigi/binfmt --install arm64,amd64
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply buildspec.dev.yml:13]: The same steps are needed for other configurations too. My bad; I committed but didn't push the changes to remote.

**Review by utsabc — APPROVED** [review]

**Review by koladilip — APPROVED** [review]

**Pull request merged**